### PR TITLE
Migrate Post Visibility Test to Playwright

### PR DESCRIPTION
**Pull request by pooja-muchandikar**
## What?

Part of https://github.com/WordPress/gutenberg/issues/38851.
Migrate[ post-visibility.test.js ](https://github.com/WordPress/gutenberg/blob/trunk/packages/e2e-tests/specs/editor/various/post-visibility.test.js) to its Playwright version.
## Why?

Part of https://github.com/WordPress/gutenberg/issues/38851.
## How?

See[ MIGRATION.md](https://github.com/WordPress/gutenberg/blob/trunk/test/e2e/MIGRATION.md) for migration steps.
## Testing Instructions

Run` npm run test-e2e:playwright test/e2e/specs/editor/various/post-visibility.spec.js`

### CODE REVIEW

**Comment by pooja-muchandikar** [comment]
Hi @kevin940726,

I hope you are going great!..

I worked on this PR and the CI is failing multiple times.. I tried fixing the issue and in local the test cases seems to be executing successfully, but CI is failing.. Could you suggest here please?

image

File: test/e2e/specs/editor/various/post-visibility.spec.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+		await editor.openDocumentSettingsSidebar();
+
+		// Set a publish date for the next month.
+		await page.click(
+			'.components-button.edit-post-post-schedule__toggle.is-tertiary'
+		);
+
+		await page.click(
+			'role=button[name="Move forward to switch to the next month."i]'
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/post-visibility.spec.js]: I think the name should be `View next month`. Not sure why it's showing differently on your machine? Maybe a rebase would help?

Comment (pooja-muchandikar) [reply test/e2e/specs/editor/various/post-visibility.spec.js]: @kevin940726, Rebased the code but still showing the same name. 

image

Comment (kevin940726) [reply test/e2e/specs/editor/various/post-visibility.spec.js]: I think the name has changed since https://github.com/WordPress/gutenberg/pull/43005. Did you forget to rerun `npm run build` or `npm run dev` after rebasing?

Comment (pooja-muchandikar) [reply test/e2e/specs/editor/various/post-visibility.spec.js]: @kevin940726,

Updated the test case as per the feedbacks. Please check..

**Comment by pooja-muchandikar** [comment]
Hi @kevin940726,

Are there any further suggestions/feedbacks on this PR?

**Comment by pooja-muchandikar** [comment]
Hi @kevin940726,

I hope you are doing great!

I wanted to check with you if there are any further feedbacks on this PR?

File: test/e2e/specs/editor/various/post-visibility.spec.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Post visibility', () => {
+	const wp = '';
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/post-visibility.spec.js]: Why is this needed?

File: test/e2e/specs/editor/various/post-visibility.spec.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+	test.afterEach( async ( { pageUtils } ) => {
+		await pageUtils.setBrowserViewport( 'large' );
+	} );
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/post-visibility.spec.js]: Since we're recreating the `page` instance for every test, I think we don't need to reset it after each test. Let's try removing it?

```suggestion
```

File: test/e2e/specs/editor/various/post-visibility.spec.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+	test( 'visibility remains private even if the publish date is in the future', async ( {
+		page,
+		admin,
+		editor,
+	} ) => {
+		await admin.createNewPost();
+
+		// Enter a title for this post.
+		await page.type( '.editor-post-title__input', 'Title' );
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/post-visibility.spec.js]: Please use role selectors for this.

**Review by kevin940726 — COMMENTED** [review]
Thanks for your patience! Left some minor feedback.

File: test/e2e/specs/editor/various/post-visibility.spec.js (hunk 2 above)
Comment (pooja-muchandikar) [reply test/e2e/specs/editor/various/post-visibility.spec.js:7]: We are using this variable below to get the status of the posts. So it's just declaration of that variable.

Comment (pooja-muchandikar) [reply test/e2e/specs/editor/various/post-visibility.spec.js]: Hi @kevin940726,

We are using this variable below to get the status of the posts. So it's just declaration of that variable.

**Comment by pooja-muchandikar** [comment]
Hi @kevin940726,

I have addressed all the feedbacks, please check and let me know if there are any further feedbacks.

Thanks!

File: test/e2e/specs/editor/various/post-visibility.spec.js (hunk 2 above)
Comment (kevin940726) [reply test/e2e/specs/editor/various/post-visibility.spec.js]: I believe we can just call `window.wp` instead. Will that work?

Comment (pooja-muchandikar) [reply test/e2e/specs/editor/various/post-visibility.spec.js]: Checking on that..

Comment (pooja-muchandikar) [reply test/e2e/specs/editor/various/post-visibility.spec.js]: This is done

**Comment by pooja-muchandikar** [comment]
Hi @kevin940726,

All feedbacks are addressed, but the CI is failing and its not related to the test case I worked upon. `End-to-End Tests / Puppeteer ` is failing, could you please check?

image

**Review by kevin940726 — APPROVED** [review]
LGTM 👍 Thanks!
